src/useOverlay.tsx

```diff
@@ -0,0 +1,68 @@
+import { RefObject, useEffect } from 'react';
+
+export function useOverlay(
+  rootElementId: string,
+  isOpen: boolean,
+  close: () => void,
+  ref: RefObject<HTMLDivElement>
+): void {
+  useEffect(() => {
+    if (!isOpen) {
+      return;
+    }
+
+    const handleKeydown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        close();
+        return;
+      }
+
+      if (event.key === 'Tab') {
+        event.preventDefault();
+
+        if (ref.current === null) {
+          return;
+        }
+
+        // https://gomakethings.com/how-to-get-the-first-and-last-focusable-elements-in-the-dom/
+        const focusables: HTMLElement[] = Array.from(
+          ref.current.querySelectorAll(
+            'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'
+          )
+        );
```

Comment:
```suggestion
        // https://gomakethings.com/how-to-get-the-first-and-last-focusable-elements-in-the-dom/
        // https://html.spec.whatwg.org/multipage/interaction.html#the-tabindex-attribute
        const focusables: HTMLElement[] = Array.from(
          ref.current.querySelectorAll(
            'button, a[href], input:not([type="hidden"]), select, details > summary:first-child, textarea, [tabindex]:not([tabindex="-1"]), [contenteditable]'
          )
        );
```

In my opinion: changed to the elements described in [HTML Living Standard](https://html.spec.whatwg.org/multipage/interaction.html#the-tabindex-attribute)

Reply:
Thank you!